From 048fe618e9619dcc73f7d47799169f5c396baf58 Mon Sep 17 00:00:00 2001
From: Jose Dapena Paz <jdapena@igalia.com>
Date: Tue, 23 Jul 2024 22:24:02 +0200
Subject: GCC: provide default values for syncer::ModelTypeInfo

GCC fails to initialize an std::array of ModelTypeInfo because the
default constructor is deleted. This is GCC behavior that, for structs
without default values, do not provide that.

Added default values for all members.

Bug: 41455655
Change-Id: I701d18643168fadf767e28b741d7b5c97c9c0e3e
---
 components/sync/base/model_type.cc | 13 +++++++------
 1 file changed, 7 insertions(+), 6 deletions(-)

diff --git a/components/sync/base/model_type.cc b/components/sync/base/model_type.cc
index 90b265e7f5ab0..c3c6232805db9 100644
--- a/components/sync/base/model_type.cc
+++ b/components/sync/base/model_type.cc
@@ -16,22 +16,23 @@ namespace syncer {
 namespace {
 
 struct ModelTypeInfo {
-  const ModelType model_type;
+  const ModelType model_type = UNSPECIFIED;
   // Used to identify the model type in the SyncModelType histogram_suffix in
   // histograms.xml. Must always be kept in sync.
-  const char* const histogram_suffix;
+  const char* const histogram_suffix = nullptr;
   // Root tag for Model Type
   // This should be the same as the model type but all lowercase.
-  const char* const lowercase_root_tag;
+  const char* const lowercase_root_tag = nullptr;
   // String value for Model Type
   // This should be the same as the model type but space separated and the
   // first letter of every word capitalized.
-  const char* const model_type_debug_string;
+  const char* const model_type_debug_string = nullptr;
   // Field number of the model type specifics in EntitySpecifics.
-  const int specifics_field_number;
+  const int specifics_field_number = -1;
   // Model type value from SyncModelTypes enum in enums.xml. Must always be in
   // sync with the enum.
-  const ModelTypeForHistograms model_type_histogram_val;
+  const ModelTypeForHistograms model_type_histogram_val =
+      ModelTypeForHistograms::kUnspecified;
 };
 
 // Below struct entries are in the same order as their definition in the
-- 
2.43.0

